### ISSUE TITLE
Patchdeps.bat not working

### ISSUE THREAD

**Issue by sindhuja07** [Microsoft/Pyjion#204]
I was trying to run PatchDeps.bat file,but it gives patch cannot be applied error.

C:\Users\gillelas\Desktop\Pyjion>PatchDeps.bat
Disabling COM interop support in CoreCLR...
error: patch failed: src/inc/utilcode.h:8
error: src/inc/utilcode.h: patch does not apply
error: patch failed: src/utilcode/CMakeLists.txt:134
error: src/utilcode/CMakeLists.txt: patch does not apply
error: patch failed: src/utilcode/longfilepathwrappers.cpp:1190
error: src/utilcode/longfilepathwrappers.cpp: patch does not apply
error: patch failed: src/utilcode/util.cpp:212
error: src/utilcode/util.cpp: patch does not apply
error: patch failed: src/CMakeLists.txt:125
error: src/CMakeLists.txt: patch does not apply
error: patch failed: build.cmd:338
error: build.cmd: patch does not apply
error: patch failed: clr.coreclr.props:10
error: clr.coreclr.props: patch does not apply
error: patch failed: clr.defines.targets:18
error: clr.defines.targets: patch does not apply
error: patch failed: clr.desktop.props:19
error: clr.desktop.props: patch does not apply
error: patch failed: CMakeLists.txt:762
error: CMakeLists.txt: patch does not apply
Applying changes to enable JIT support in Python...
error: patch failed: Include/ceval.h:10
error: Include/ceval.h: patch does not apply
error: patch failed: Include/code.h:7
error: Include/code.h: patch does not apply
error: patch failed: Include/dictobject.h:142
error: Include/dictobject.h: patch does not apply
error: patch failed: Include/pystate.h:12
error: Include/pystate.h: patch does not apply
error: patch failed: Objects/codeobject.c:152
error: Objects/codeobject.c: patch does not apply
error: patch failed: Python/ceval.c:770
error: Python/ceval.c: patch does not apply
..\Patches\Python\Python\pylifecycle.c:23: trailing whitespace.
#if defined(WIN32) || defined(WIN64)
..\Patches\Python\Python\pylifecycle.c:31: trailing whitespace.

..\Patches\Python\Python\pylifecycle.c:39: trailing whitespace.
#if defined(WIN32) || defined(WIN64)
error: patch failed: Python/pylifecycle.c:16
error: Python/pylifecycle.c: patch does not apply
error: patch failed: Python/pystate.c:80
error: Python/pystate.c: patch does not apply

I also tried applying patches one by one by going into the directory by using "git apply" on individual files, it doesnt work either. It shows something like this:

C:\Users\gillelas\Desktop\Pyjion\CoreCLR>git apply ..\Patches\CoreCLR\src\utilcode\longfilepathwrappers.cpp --verbose
Checking patch src/utilcode/longfilepathwrappers.cpp...
error: while searching for:

#ifndef FEATURE_PAL

#if ! defined(DACCESS_COMPILE) && !defined(SELF_NO_HOST)
extern HINSTANCE            g_pMSCorEE;
#endif// ! defined(DACCESS_COMPILE) && !defined(SELF_NO_HOST)

BOOL PAL_GetPALDirectoryWrapper(SString& pbuffer)
{

error: patch failed: src/utilcode/longfilepathwrappers.cpp:1190
error: src/utilcode/longfilepathwrappers.cpp: patch does not apply

I wanted to know why this is happening. Any help regarding this would be greatly appreciated.

**Comment by brettcannon** [comment]
Did you make sure to do a `git clone --recursive` for the checkout?

**Comment by sindhuja07** [comment]
Yeah i did a recursive clone from git. Inspire of that I get this strange
error

**Comment by brettcannon** [comment]
Then the source code has sufficiently changed for the patches to no longer apply cleanly (since we don't check out fresh code regularly we sometimes don't notice when this happens).

Once Python 3.6 is released we will need to update Pyjion for that, but a perk is it won't necessitate patching Python itself anymore and so we can drop those patches. We will also re-evaluate our need to patch RyuJIT (and hopefully drop that as well).

**Comment by sindhuja07** [comment]
Hello,

One more doubt. I was trying to run the benchmark suite I downloaded from
https://hg.python.org/benchmarks/file/tip .

I used this command
 py -3 perf.py -r --csv ..\Pyjion5\Perf\2016-05-13.csv
--benchmarks=all,-chameleon_v2,-hexiom2
C:\users\gillelas\desktop\Pyjion\Python\PCbuild\nojit\python.exe
C:\users\gillelas\desktop\Pyjion\Python\PCbuild\amd64\python.exe, as listed
here:

https://github.com/Microsoft/Pyjion/tree/master/Perf

 but I dont see any nojit folder inside PCBuild and also there is no
python.exe inside amd64 folder.

Can you please help me run these benchmarks with perf? I will be waiting to
hear back from you.

**Comment by brettcannon** [comment]
I assume @DinoV just changed a build target to get Python built into a new directory. But in any case it doesn't matter as that's no longer Python's official benchmark suite, https://github.com/python/performance is (if you really care about benchmarking against the old benchmarks you can check Pyjion out twice and simply build Python on its own in one of the checkouts for a baseline interpreter).

But (a) I'm on holiday, and (b) development on Pyjion is frozen until Python 3.6.0 is released and we have the time to update the project to work with that version of Python. I have updated our README to state this thanks to this issue.

**Comment by tonybaloney** [comment]
Implemented in #237 (No longer required to patch deps)
